MdeModulePkg/Bus/Pci/CxlDxe/CxlDxe.h

```diff
@@ -0,0 +1,362 @@
+/** @file
+  Header file for CxlDxe driver
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#ifndef _EFI_CXLDXE_H_
+#define _EFI_CXLDXE_H_
+
+#include <string.h>
+#include <Protocol/PciIo.h>
+#include <Protocol/FirmwareManagement.h>
+#include <IndustryStandard/Pci.h>
+#include <IndustryStandard/Cxl20.h>
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/DebugLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+#include <Library/UefiLib.h>
+#include <Library/UefiRuntimeServicesTableLib.h>
+
+#define CXL_MEMORY_CLASS                              0x05
+#define CXL_MEMORY_SUB_CLASS                          0x02
+#define CXL_MEMORY_PROGIF                             0x10
+#define CXL_PCIE_EXTENDED_CAP_NEXT(n)                 ((n) >> (CXL_PCIE_EXTENDED_NEXT_CAP_OFFSET_SHIFT))
+#define CXL_IS_DVSEC(n)                               (((n) & (0xFFFF)) == CXL_PCI_DVSEC_VENDOR_ID)
+#define CXL_CONTROLLER_PRIVATE_DATA_SIGNATURE         SIGNATURE_32 ('C','X','L','X')
+#define CXL_PCI_EXT_CAP_ID(header)                    (header & 0x0000ffff)
+#define CXL_PCI_EXT_CAP_NEXT(header)                  ((header >> 20) & 0xffc)
+#define CXL_DEV_CAP_ARRAY_OFFSET                      0x0
+#define CXL_DEV_CAP_ARRAY_CAP_ID                      0
+#define CXL_BIT(nr)                                   ((UINT32)1 << nr)
+#define CXL_DEV_MBOX_CTRL_DOORBELL                    CXL_BIT(0)
+#define CXL_SZ_1M                                     0x00100000
+#define CXL_STRING_BUFFER_WIDTH                       256
+#define CXL_FW_IMAGE_ID                               1
+#define CXL_FW_VERSION                                1
+#define CXL_PACKAGE_VERSION_FFFFFFFE                  0xFFFFFFFE
+#define CXL_FIRMWARE_IMAGE_ID_NAME                    L"CXL Firmware Version 1.0"
+#define CXL_PACKAGE_VERSION_NAME                      L"CXL Firmware Package Version Name UEFI Driver"
+#define CXL_FW_SIZE                                   32768    /* 32 mb */
+#define CXL_BITS_PER_LONG                             32
+#define CXL_UL                                        (UINTN)
+#define CXL_GENMASK(h, l)                             (((~CXL_UL(0)) - (CXL_UL(1) << (l)) + 1) & (~CXL_UL(0) >> (CXL_BITS_PER_LONG - 1 - (h))))
+#define CXL_CONTROLLER_PRIVATE_FROM_FIRMWARE_MGMT(a)  CR (a, CXL_CONTROLLER_PRIVATE_DATA, FirmwareMgmt, CXL_CONTROLLER_PRIVATE_DATA_SIGNATURE)
+#define CXL_QEMU                                      1
+#define CXL_FW_REVISION_LENGTH_IN_BYTES               16
+
+typedef struct {
+  UINT16    VendorID;
+  UINT16    DeviceID;
+} CXL_VENDORID;
+
+typedef enum {
+  PCIE_EXT_CAP_HEADER,
+  PCIE_DVSEC_HEADER_1,
+  PCIE_DVSEC_HEADER_2,
+  PCIE_DVSEC_HEADER_MAX
+} CXL_PCIE_DVSEC_HEADER_ENUM;
+
+/* Register Block Identifier (RBI) */
+enum cxl_regloc_type {
```

Comment:
The enum, struct type, and field names below do not follow code style.  Are any of these from the CXL specs?  Should they be moved to MdePkg/Include/IndustryStandard/Cxl* files?

Reply:
Review comment is taken care in updated patch.